tests/cpp/test_gpu_view.cpp

```diff
@@ -2761,4 +2761,35 @@ TEST_F(GpuViewTest, FusionMismatchingReshape) {
   testValidate(&fusion, cg_outputs, {t0}, __LINE__, __FILE__);
 }
 
+// Test that replacement of a scalar in a reshaped ID that is a product of a
+// split gets replaced properly
```

Comment:
Can you please add more details about where the problem occurs and how it could manifest if not fixed? Is it also possible to do more explicit validation of the fix?

Reply:
I expanded this comment and also added a test where I explicitly do the replacement and check that it works as expected. I kept this original test since it is the repro for the bug.